lib/elixir/lib/kernel.ex

```diff
@@ -261,6 +261,8 @@ defmodule Kernel do
   @doc """
   Returns the head of a list. Raises `ArgumentError` if the list is empty.
 
+  It works with improper lists.
```

Comment:
@michalmuskala not sure I agree with that wording. Almost no function does that as a checking, it is just a question if it traverses to the end or not really.